sentry/src/main/java/io/sentry/clientreport/ClientReportRecorderImpl.java

```diff
@@ -0,0 +1,176 @@
+package io.sentry.clientreport;
+
+import io.sentry.DateUtils;
+import io.sentry.SentryEnvelope;
+import io.sentry.SentryEnvelopeItem;
+import io.sentry.SentryItemType;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.transport.DataCategory;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class ClientReportRecorderImpl implements ClientReportRecorder {
+
+  private static volatile @Nullable ClientReportRecorderImpl SHARED_INSTANCE = null;
+
+  public static @NotNull ClientReportRecorderImpl getInstance() {
```

Comment:
Passing `SentryOptions` here would make the interface methods a bit cleaner but I'm not sure I like passing the options to the `getInstance` method since only the first invocation will have an effect here. That's why I preferred to have the options in all the methods instead of the constructor.

Reply:
I believe it'd be fine passing it via ctor, for example, See `JsonSerializer`.

Reply:
But that's not a singleton where only the options of the first invocation are used.

Reply:
Exactly, I meant that `ClientReportRecorderImpl` does not need to be a singleton, if the instance of the class lives in the SentryOptions, unless that multi-thread systems have a new options instance per hub, not sure if its the case.

Reply:
Changed to single instance on options, options are passed to ctor now